python/pyspark/ml/tests.py

```diff
@@ -657,32 +656,104 @@ def test_logistic_regression(self):
         except OSError:
             pass
 
+    def _compare_params(self, m1, m2, param):
```

Comment:
This can be used in a lot of other places below (everywhere you compare 2 param values & parents).